src/OrleansTestingHost/TestingSiloHost.cs

```diff
@@ -557,14 +563,20 @@ private SiloHandle StartOrleansSilo(Silo.SiloType type, TestingSiloOptions optio
 
             if (nodeConfig.ProxyGatewayEndpoint != null && nodeConfig.ProxyGatewayEndpoint.Address != null)
             {
-                nodeConfig.ProxyGatewayEndpoint = new IPEndPoint(nodeConfig.ProxyGatewayEndpoint.Address, ProxyBasePort + instanceCount);
+                int proxyBasePort = options.ProxyBasePort < 0 ? ProxyBasePort : options.ProxyBasePort;
+                nodeConfig.ProxyGatewayEndpoint = new IPEndPoint(nodeConfig.ProxyGatewayEndpoint.Address, proxyBasePort + instanceCount);
             }
 
             config.Globals.ExpectedClusterSize = 2;
 
             config.Overrides[siloName] = nodeConfig;
 
-            AdjustForTest(config);
+            if (host != null)
+                host.AdjustForTest(config);
+
+            if (options.ConfigurationCustomizer != null)
```

Comment:
why do we need the 2 ways to Adjust/Customize Configuration For Test? The host.AdjustForTest is not enough?